### PR TITLE
Add datadog client library

### PR DESCRIPTION
**Pull request by brentmmarks**
Have extracted this out of old code from nomad-scaler used to pull metrics from Datadog. This will be used to compare info with Nomad and Docker versions for SLOs around time since latest upgrade.

### CODE REVIEW

File: datadog/client_test.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+	u, _ := url.Parse(ts.URL)
+	u.Path = "/api/v1/"
+
+	client = &Client{
+		APIKey:  "API Key",
+		AppKey:  "Application Key",
+		BaseURL: u,
+	}
+	cleanup = func() {
```
Comment (pete-woods) [inline datadog/client_test.go:356]: Should probably make this a t.Cleanup now

**Review by pete-woods — APPROVED** [review]

File: datadog/client_test.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+				_, _ = fmt.Fprintln(w, `{garbage JSON}`)
+			},
+			wantValid: false,
+			wantErr:   "failed to decode error",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (pete-woods) [inline datadog/client_test.go:71]: We're not doing parallel tests any more

File: datadog/client_test.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+				}`)
+			},
+			wantErr:  "failed to decode data point",
+			wantResp: emptySeries,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (pete-woods) [inline datadog/client_test.go:310]: No more parallel

**Review by pete-woods — APPROVED** [review]
Might be nice to port this over to the ex HTTP client after merging (if you have time)

**Comment by Dustinturner44** [comment]
```

```

**Pull request merged**